Update search.search() page (#35858)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/search/search/index.md

A search using Wikipedia with the results shown in a new window:

```js

> async function search() {
>   try {
>     // this will fail if `Wikipedia (en)` is not an engine of the browser
>     await browser.search.search({
>       query: "styracosaurus",
>       engine: "Wikipedia (en)",
>       disposition: "NEW_WINDOW",
>     });
>   } catch (ex) {
>     // initiate a search manually for fallback
>     await browser.windows.create({
>       url: "https://en.wikipedia.org/w/index.php?title=Special:Search&search=styracosaurus",
>     });
>   }

Reviewer: I would suggest updating the code comments:
```suggestion
async function search() {
  try {
    // try to search using the `Wikipedia (en)` search engine
    await browser.search.search({
      query: "styracosaurus",
      engine: "Wikipedia (en)",
      disposition: "NEW_WINDOW",
    });
  } catch (ex) {
    // if the search fails, e.g., because the search engine isn't defined to the browser, initiate the search using a url
    await browser.windows.create({
      url: "https://en.wikipedia.org/w/index.php?title=Special:Search&search=styracosaurus",
    });
  }
```

A search using Wikipedia with the results shown in a new window:

```js
async function search() {
  try {

>     // This will fail, if the user has removed the search engine
>     // or has other version of the search engine installed (e.g. Wikipedia (de)).

Reviewer 2: having `Wikipedia (de)` and `Wikipedia (en)` will not fail the function. additionally, not all browser has `Wikipedia (en)` bundled with them - `removed` may not be accurate.

```suggestion
    // this will fail if `Wikipedia (en)` is not an engine of the browser
```

Author: > having `Wikipedia (de)` and `Wikipedia (en)` will not fail the function.

I mean, if the user **only** has `Wikipedia (de)` installed then it **will** fail!

Reviewer 2: > > having `Wikipedia (de)` and `Wikipedia (en)` will not fail the function.
> 
> I mean, if the user **only** has `Wikipedia (de)` installed then it **will** fail!

then what you actually wrote in your code controverts with what you means here. you just want to say **`Wikipedia (en)` is not a search engine of the browser**.

    await browser.search.search({
      query: "styracosaurus",
      engine: "Wikipedia (en)",
      disposition: "NEW_WINDOW",
    });
  } catch (ex) {

>     // Initiate a search manually

Reviewer 2: ```suggestion
    // initiate a search manually for fallback
```
